lib/ramble/ramble/application.py

```diff
@@ -1419,9 +1443,168 @@ def format_context(context_match, context_format):
                     fom_copy['name'] = fom_name
                     context_map['foms'].append(fom_copy)
 
-                results['CONTEXTS'].append(context_map)
+                self.results['CONTEXTS'].append(context_map)
+
+        workspace.append_result(self.results)
+
+    def calculate_statistics(self, workspace):
+        """Calculate statistics for results of repeated experiments
+
+        When repeated experiments are used, this method aggregates the results of
+        each experiment's repeats and calculates statistics for each numeric FOM.
+
+        If a FOM is non-numeric, no calculations are performed.
+
+        Statistics are injected into the results file under the base experiment
+        namespace.
+        """
+
+        def is_numeric(value):
+            """Returns true if a fom value is numeric"""
+
+            try:
+                float(value)
+                return True
+            except ValueError:
+                return False
+
+        repeat_experiments = {}
+        repeat_foms = {}
+        first_repeat_exp = ''
+
+        # repeat_experiments dict {repeat experiment namespace: {dict}}
+        # repeat_foms dict {context: {(fom name, units, origin, origin_type): [list of values]}}
+        # origin_type is generated as 'summary::stat_name'
+
+        # Check if this is the base experiment of a set of repeats
+        if not self.repeats[0]:
+            return
+        n_repeats = self.repeats[1]
+
+        base_exp_name = self.expander.experiment_name
+        base_exp_namespace = self.expander.experiment_namespace
+
+        # Create a list of all repeats by inserting repeat index
+        for n in range(1, n_repeats + 1):
+            if '.chain' in base_exp_name:
```

Comment:
Changed this to look at membership in the experiment_set.chained_experiments and experiment_set.experiments instead of parsing the name. If it's in the chained experiment list, and NOT in the experiment list, it will have the '.chain' in the name. For non-chained experiments, the ``base_exp_name`` is found in both lists (as part of the name of chained experiments)